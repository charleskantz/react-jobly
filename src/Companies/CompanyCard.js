import React from "react";
import { Card } from '../Common/Card';
import { Image } from '../Common/Image';
import { ButtonLink } from '../Common/StyledLink';
import { Heading, Body, LightMiceType } from '../Common/Type';
import styled from '@emotion/styled';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBriefcase, faUsers } from '@fortawesome/free-solid-svg-icons';

export const DetailsDiv = styled.div`
  margin-left: 16px;
`;

const Container = styled.div`
  display: flex;
`;

const JobsCTA = styled.div`
  padding: 8px 8px 8px 16px;
  border: 1px solid rgb(228, 231, 240);
  border-radius: 4px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 16px;

  &:hover {
    border: 1px solid #acbdd5;
  }
`;

/** CompanyCard - display component with company details
 *
 * @param {company} prop: company details
 */
function CompanyCard({ company }) {

  // separate details for easier reading
  const { handle, name, description, num_employees, logo_url, job_count } = company;

  return (
    <Card column>
      <Container>
        <Image
          src={logo_url}
          size={72}
          alt={`logo for ${name}`}
        >
        </Image>
        <DetailsDiv>
          <Heading>{name}</Heading>
          <Body>{description}</Body>
          <LightMiceType>
            <FontAwesomeIcon icon={faUsers} /> {num_employees} EMPLOYEES
          </LightMiceType>
        </DetailsDiv>
      </Container>
      <JobsCTA>
        <Body>
          <FontAwesomeIcon icon={faBriefcase} /> {job_count} Jobs Available
        </Body>
        <ButtonLink href={`/companies/${handle}`}>See Jobs</ButtonLink>
      </JobsCTA>
    </Card>
  )
}

export default CompanyCard;

//       <Link style={{textDecoration: "none", color:"black"}} className="Company" to={`/companies/${handle}`}>
