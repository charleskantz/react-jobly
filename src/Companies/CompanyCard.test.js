import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";

const companyMock = {
  handle: "testinc",
  name: "Test Inc",
  description: "Testing test 1 2 3 test",
  num_employees: 1,
  logo_url: null,
  job_count: 5
}

// smoke
it("renders without crashing", function() {
  render(
    <CompanyCard company={companyMock} />
  );
});

// snapshot
it("matches snapshot", function() {
  const { asFragment } = render(<CompanyCard company={companyMock} />);
  expect(asFragment()).toMatchSnapshot();
});