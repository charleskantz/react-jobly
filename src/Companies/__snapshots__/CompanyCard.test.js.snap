// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-omwz41"
  >
    <div
      class="css-ov1ktg"
    >
      <img
        alt="logo for Test Inc"
        class="css-nci47p"
        size="72"
      />
      <div
        class="css-ci461p"
      >
        <h1
          class="css-1awjs8x"
        >
          Test Inc
        </h1>
        <p
          class="css-zsarox"
        >
          Testing test 1 2 3 test
        </p>
        <p
          class="css-1s86eya"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-users fa-w-20 "
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
           1 EMPLOYEES
        </p>
      </div>
    </div>
    <div
      class="css-1f9d4lu"
    >
      <p
        class="css-zsarox"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-briefcase fa-w-16 "
          data-icon="briefcase"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
            fill="currentColor"
          />
        </svg>
         5 Jobs Available
      </p>
      <a
        class="css-9tqi29"
        href="/companies/testinc"
      >
        See Jobs
      </a>
    </div>
  </div>
</DocumentFragment>
`;
